forcing_tools/netcdf2vic.py

```diff
@@ -0,0 +1,372 @@
+
+
+# -------------------------------------------------------------------- #
+def process_command_line():
+    """
+    Parse arguments and assign flags for further loading of variables, for
+    information on input arguments, type netcdf2vic.py -h
+    """
+    # Parse arguments
+    parser = argparse.ArgumentParser()
+    parser.add_argument("config", type=str, help="Input Configuration File")
+    args = parser.parse_args()
+
+    config_dict = read_config(args.config)
+
+    return config_dict
+
+
+# -------------------------------------------------------------------- #
+# Read the Configuration File
+def read_config(config_file):
+    """
+    Return a dictionary with subdictionaries of all configFile options/values
+    """
+    config = ConfigParser.SafeConfigParser()
+    config.optionxform = str
+    config.read(config_file)
+    sections = config.sections()
+    dict1 = OrderedDict()
+    for section in sections:
+        options = config.options(section)
+        dict2 = OrderedDict()
+        for option in options:
+            dict2[option] = config_type(config.get(section, option))
+        dict1[section] = dict2
+    return dict1
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+# Find the type of the config options
+def config_type(value):
+    """
+    Parse the type of the configuration file option.
+    First see the value is a bool, then try float, finally return a string.
+    """
+    val_list = [x.strip() for x in value.split(',')]
+    if len(val_list) == 1:
+        value = val_list[0]
+        if value in ['true', 'True', 'TRUE', 'T']:
+            return True
+        elif value in ['false', 'False', 'FALSE', 'F']:
+            return False
+        elif value in ['none', 'None', 'NONE', '']:
+            return None
+        elif isint(value):
+            return int(value)
+        elif isfloat(value):
+            return float(value)
+        else:
+            return os.path.expandvars(value)
+    else:
+        try:
+            return map(float, val_list)
+        except:
+            pass
+        try:
+            return map(int, val_list)
+        except:
+            return val_list
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def isfloat(x):
+    """Test of value is a float"""
+    try:
+        a = float(x)
+    except ValueError:
+        return False
+    else:
+        return True
+# -------------------------------------------------------------------- #
+
+
+# -------------------------------------------------------------------- #
+def isint(x):
+    """Test if value is an integer"""
+    try:
+        a = float(x)
+        b = int(a)
+    except ValueError:
+        return False
+    else:
+        return a == b
+# -------------------------------------------------------------------- #
```

Comment:
Do we really need these functions? In Python 3 (and maybe 2 as well) you can just use `isinstance`, e.g. `isinstance(x, int)`